### PR TITLE
use `dist/release` KEYS

### PR DESCRIPTION
**Pull request by kevinjqliu**
Deprecate the use of 
- https://dist.apache.org/repos/dist/dev/iceberg/KEYS
in favor of 
- https://downloads.apache.org/iceberg/KEYS which is an alias for https://dist.apache.org/repos/dist/release/iceberg/KEYS

devlist discussion: https://lists.apache.org/thread/8j41w4y2jx6r3ybj0o82bfyn0npmhgx2

### CODE REVIEW

**Review by zeroshade — APPROVED** [review]
LGTM

File: dev/release/verify_rc.sh [hunk 1]
```diff
@@ -34,7 +34,7 @@ set -x
 VERSION="$1"
 RC="$2"
 
-ICEBERG_DIST_BASE_URL="https://dist.apache.org/repos/dist/dev/iceberg"
+ICEBERG_DIST_BASE_URL="https://dist.apache.org/repos/dist/release/iceberg"
```
Comment (RussellSpitzer) [inline dev/release/verify_rc.sh]: Shouldn't this use the "downloads" location?

Comment (RussellSpitzer) [reply dev/release/verify_rc.sh]:
```sh
import_gpg_keys() {
  if [ "${VERIFY_SIGN}" -gt 0 ]; then
    download "${ICEBERG_DIST_BASE_URL}/KEYS"
    gpg --import KEYS
  fi
}
```

Is used below

Comment (kevinjqliu) [reply dev/release/verify_rc.sh]: I thought https://downloads.apache.org/iceberg/KEYS and https://dist.apache.org/repos/dist/release/iceberg/KEYS were the same file, and that one is an alias of another. 
But it actually looks like they are different. For example, a search for "Matt Topol" only appears in the `dist/release` KEYS

Comment (kevinjqliu) [reply dev/release/verify_rc.sh]: nevermind they are the same...

Comment (kevinjqliu) [reply dev/release/verify_rc.sh]: https://downloads.apache.org/iceberg/KEYS is a mirror of https://dist.apache.org/repos/dist/release/iceberg/KEYS

and might be delayed when updating

**Comment by kevinjqliu** [comment]
Updated to use `downloads` whenever possible, except when uploading new keys

**Review by Fokko — APPROVED** [review]

**Review by RussellSpitzer — APPROVED** [review]

**Comment by RussellSpitzer** [comment]
Thanks @kevinjqliu for the PR and @zeroshade, @Fokko for review!

**Pull request merged**